Automattic/woocommerce-payments
# Send metadata in the error message

#### Changes proposed in this Pull Request
In this PR we append the `intent_meta_order_id` and `order_id` to the error message to skip changes in WC.

Conversation from https://github.com/Automattic/woocommerce-payments/pull/7873.
P2: pdpOw2-45Y-p2

#### Testing instructions
1. Test with WooPay https://github.com/Automattic/woopay/pull/2413
2. Update the [condition](https://github.com/Automattic/woocommerce-payments/blob/0d3bd80bca2f10a6a75b35b01d0f6004bc88436e/includes/class-wc-payment-gateway-wcpay.php#L1273) to `true`
3. Add and log below code before [this line](https://github.com/Automattic/woopay/blob/c36461855d734ae448001eb9e145c8c73ed6f270/src/Tracking/OrderStatusTracker.php#L57C3-L57C3)
```
$previous = $exception->getPrevious();
$meta     = $previous instanceof MetaException ? $previous->getMessage() : '';

error_log( sprintf( 'Meta: %s', $meta ) );
```
5. Place an order via WooPay
6. Check the meta in the debug log, should see `intent_meta_order_id` and `order_id`

-------------------


**Post merge**
